### PR TITLE
feat: add permission for Item content creator role

### PR DESCRIPTION
**Pull request by chinnu-focaloid**
Relates to https://oat-sa.atlassian.net/browse/ADF-208

### CODE REVIEW

File: migrations/Version202107271111381465_qtiItemPci.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+namespace oat\qtiItemPci\migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use oat\tao\scripts\tools\migrations\AbstractMigration;
+use oat\tao\scripts\tools\accessControl\SetRolesAccess;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version202107271111381465_qtiItemPci extends AbstractMigration
```
Comment (gabrielfs7) [inline migrations/Version202107271111381465_qtiItemPci.php:14]:
```suggestion
final class Version202107271111381465_qtiItemPci extends AbstractMigration
```

File: migrations/Version202107271111381465_qtiItemPci.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+{
+
+    private const CONFIG = [
+        SetRolesAccess::CONFIG_RULES => [
+            'http://www.tao.lu/Ontologies/TAOItem.rdf#ItemContentCreatorRole' => [
```
Comment (gabrielfs7) [inline migrations/Version202107271111381465_qtiItemPci.php:19]: Can we use the constant for `http://www.tao.lu/Ontologies/TAOItem.rdf#ItemContentCreatorRole`?

Comment (chinnu-focaloid) [reply migrations/Version202107271111381465_qtiItemPci.php:19]: Can I use the constant defined on taoItems/models/classes/user/TaoItemsRoles.php

Comment (gabrielfs7) [reply migrations/Version202107271111381465_qtiItemPci.php:19]: I think yes, the place where you have this constant.

This extension depends on itemqti that depends on items extension: https://github.com/oat-sa/extension-tao-itemqti-pci/blob/master/composer.json#L27

Comment (chinnu-focaloid) [reply migrations/Version202107271111381465_qtiItemPci.php]: Done

File: migrations/Version202107271111381465_qtiItemPci.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                '--' . SetRolesAccess::OPTION_CONFIG,
+                self::CONFIG,
+            ]
+        );
+    }
+
+    public function down(Schema $schema): void
+    {
+        $this->throwIrreversibleMigrationException(
+            'This role should have been applied in the past, so we should not roll it back'
```
Comment (shpran) [inline migrations/Version202107271111381465_qtiItemPci.php:42]: What does this mean?

File: manifest.php [hunk 4]
```diff
@@ -48,6 +49,7 @@
         ['grant', 'http://www.tao.lu/Ontologies/generis.rdf#qtiItemPciManager', ['ext' => 'qtiItemPci', 'mod' => 'PciManager', 'act' => 'export']],
         ['grant', 'http://www.tao.lu/Ontologies/generis.rdf#qtiItemPciManager', ['ext' => 'qtiItemPci', 'mod' => 'PciManager', 'act' => 'import']],
         ['grant', 'http://www.tao.lu/Ontologies/generis.rdf#qtiItemPciManager', ['ext' => 'qtiItemPci', 'mod' => 'PciManager', 'act' => 'index']],
+        ['grant', TaoItemsRoles::ITEM_CONTENT_CREATOR, ['ext' => 'qtiItemPci', 'mod' => 'PciLoader', 'act' => 'load']],
```
Comment (gabrielfs7) [inline manifest.php:52]: Lets not forget to update composer.json with new versions

**Review by shpran — COMMENTED** [review]
Do not forget to update `composer.json` with new versions before merging.

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by shpran — APPROVED** [review]

File: migrations/Version202107271111381465_qtiItemPci.php [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+use Doctrine\DBAL\Schema\Schema;
+use oat\tao\scripts\tools\migrations\AbstractMigration;
+use oat\tao\scripts\tools\accessControl\SetRolesAccess;
+use oat\taoItems\model\user\TaoItemsRoles;
+
+final class Version202107271111381465_qtiItemPci extends AbstractMigration
+{
+
+    private const CONFIG = [
```
Comment (gabrielfs7) [inline migrations/Version202107271111381465_qtiItemPci.php:15]: Should we have an installation script?

**Review by gabrielfs7 — APPROVED** [review]

**Comment by AntonNeviarouski** [comment]
QA verified on http://test-chinnufrancis.playground.kitchen.it.taocloud.org:48831/tao/Main/login

**Comment by AntonNeviarouski** [comment]
QA re-verified on http://test-chinnufrancis.playground.kitchen.it.taocloud.org:48831/tao/Main/login

**Pull request merged**